gwctl/pkg/common/clients.go

```diff
@@ -93,7 +94,28 @@ func MustClientsForTest(t *testing.T, initRuntimeObjects ...runtime.Object) *K8s
 		t.Fatal(err)
 	}
 
-	fakeClient := fakeclient.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(initRuntimeObjects...).Build()
+	// extractorFunc are used to build an index for Event objects. This is
+	// required to properly mock things like the following when listing Events
+	// associated with a resource:
+	//
+	// 	fields.OneTermEqualSelector("involvedObject.kind", "Gateway")
```

Comment:
If I'm understanding correctly, this is more for just test mocks or something right? It looks like you're using this to list events. Maybe adjust the description to note that this is the best way to be able to list events that involve a specific resource?